mmdet/models/detectors/base_detr.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from abc import ABCMeta, abstractmethod
+from typing import Dict, List, Tuple, Union
+
+from torch import Tensor
+
+from mmdet.registry import MODELS
+from mmdet.structures import OptSampleList, SampleList
+from mmdet.utils import ConfigType, OptConfigType, OptMultiConfig
+from .base import BaseDetector
+
+
+@MODELS.register_module()
+class TransformerDetector(BaseDetector, metaclass=ABCMeta):
+    """Base class for Transformer-based detectors.
+
+    Transformer-based detectors use an encoder to process output features of
+    neck, then several queries interactive with the output features of encoder
+    and do the regression and classification.
+    """
+
+    def __init__(self,
+                 backbone: ConfigType,
+                 neck: OptConfigType = None,
+                 encoder_cfg: OptConfigType = None,
+                 decoder_cfg: OptConfigType = None,
+                 positional_encoding_cfg: OptConfigType = None,
+                 bbox_head: OptConfigType = None,
+                 num_query: int = 100,
+                 train_cfg: OptConfigType = None,
+                 test_cfg: OptConfigType = None,
+                 data_preprocessor: OptConfigType = None,
+                 init_cfg: OptMultiConfig = None) -> None:
+        super().__init__(
```

Comment:
add doc here following other detectors

Reply:
I viewed the code of other detectors and found that most of them write doc in the next line of `class`, but not  `__init__()`.
I haven't understand the meaning of this comment~